src/app/ApplicationTemplate.Access/ApiClients/MinimumVersion/IMinimumVersionReposiory.cs

```diff
@@ -0,0 +1,19 @@
+﻿using System;
+
+namespace ApplicationTemplate.DataAccess;
+
+/// <summary>
+/// Gets the minimum version required to use the application.
+/// </summary>
+public interface IMinimumVersionReposiory : IDisposable
```

Comment:
Should the `IDisposable` be on the implementation rather than the interface?

Reply:
Doesnt it make more sense to put it on the interface? Since it would be accessible from the service you get from GetService? Do we only want to be able to access it if we use code like: if (service is IDisposable disposable)? I don't understand why we would want it on the implementation and not the interface

Reply:
Having it on the interface means that you must implement it if you want to implement IMinimumVersionRepository. Having it on the implementation means that this specific implementation is disposable.

Reply:
But I don't really understand the point of making an implementation disposable if it's not going to be part of it's interface? I guess in this case it makes sense because it's disposed of by the base functional test, but is it a common pattern to check for if every service is disposable and the disposing them in that way?

Reply:
Well the interface should probably not dictate if the implementation is going to use disposable stuff. This is up to the implementation.

Reply:
I made a quick test to validate and it's fine if it's only on the interface. It's still automatically disposed by the host.
![image](https://github.com/nventive/UnoApplicationTemplate/assets/39710855/c77a6c83-44a1-4df1-bee6-4c10119e3741)

Reply:
So, making something disposable is more about making it disposable for the Host / test, by telling them what to dispose of when you stop the app than making it something that's usable by developer, if I understand correctly?

Reply:
> Well the interface should probably not dictate if the implementation is going to use disposable stuff. This is up to the implementation.

This service contains an observable which is always a IDisposable, so there will always be at least one thing you could dispose

Reply:
Yeah true, I'm unsure about this one.

Reply:
`IObservable` doesn't implement `IDisposable`, but even if it did, it doesn't mean that the service would always be responsible to dispose it. It really depends on how it's implemented.

Reply:
> So, making something disposable is more about making it disposable for the Host / test, by telling them what to dispose of when you stop the app than making it something that's usable by developer elsewhere in the code, if I understand correctly?

My comment was more about the intention. If you put `IDisposable` in the interface, you explicitly say that all implementations must be disposable. In your case, I don't think that's needed, because you could have an implementation that doesn't need to be disposed.